lib/termsupport.zsh

```diff
@@ -53,14 +53,27 @@ precmd_functions+=(omz_termsupport_precmd)
 preexec_functions+=(omz_termsupport_preexec)
 
 
-# Runs before showing the prompt, to update the current directory in Terminal.app
-function omz_termsupport_cwd {
-  # Notify Terminal.app of current directory using undocumented OSC sequence
-  # found in OS X 10.9 and 10.10's /etc/bashrc
-  if [[ $TERM_PROGRAM == Apple_Terminal ]] && [[ -z $INSIDE_EMACS ]]; then
-    local PWD_URL="file://$HOSTNAME${PWD// /%20}"
-    printf '\e]7;%s\a' "$PWD_URL"
-  fi
-}
+# Keep Apple Terminal.app's current working directory updated
+# Based on this answer: http://superuser.com/a/315029
+# With extra fixes to handle multibyte chars and non-UTF-8 locales
+
+if [[ "$TERM_PROGRAM" == "Apple_Terminal" ]] && [[ -z "$INSIDE_EMACS" ]]; then
+
+  # Emits the control sequence to notify Terminal.app of the cwd
+  function update_terminalapp_cwd() {
+    # Identify the directory using a "file:" scheme URL, including
+    # the host name to disambiguate local vs. remote paths.
+
+    # Percent-encode the pathname.
+    local URL_PATH=$(omz_urlencode -P $PWD)
+    [[ $? != 0 ]] && return 1
+    local PWD_URL="file://$HOST$URL_PATH"
```

Comment:
I've tested it. 

The effect is that when you're ssh'ed in to a remote machine, the remote path is still displayed, and with the correct machine name at the root of the path (when you right-click the folder to display its path), but the folder icons are displayed with "?" on them, and won't open a Finder window to that folder when you click on it. Not very useful, but it prevents the bad case of where you're on a remote machine, and there happens to be a local folder with the same path, and you click on it to open the folder but now you're looking at different places in the terminal and Finder but the UI indicators suggest they're the same place. That could cause confusion or errors if you're not aware of exactly what's going on, and assume that you're looking at the same files. (E.g. if you want to clean up some files on the remote machine, but you end up blowing them away on the local machine, or vice versa.) Super bad if you're ssh'ing from a dev to prod box.

I could see iTerm2 getting creative with ssh use to implement the open-remote-folder-in-Finder functionality. Or maybe even doing it in OMZ with some creative path detection in zsh on the remote side to convert the `file://` URL to an `afp://` one, if there's one available.

Apple's own `/etc/bashrc` includes the hostname too, so this keeps things consistent between `bash` and `zsh`.

In theory, if file sharing is turned on for the other machines, Terminal could be able to open the remote folder when requested. But I haven't been able to get this to work. Probably because AFP shares don't necessarily correspond to the root of the remote filesystem and OS X doesn't have enough info to do that mapping, and OS X (AFAIK) doesn't have the single-sign-on and conventional `\\host\c$` shares that Windows does to make that mapping easy.